api/types/resource.go

```diff
@@ -168,7 +172,7 @@ func (m *Metadata) Expiry() time.Time {
 }
 
 // SetTTL sets Expires header using realtime clock
-func (m *Metadata) SetTTL(clock clockwork.Clock, ttl time.Duration) {
+func (m *Metadata) SetTTL(clock Clock, ttl time.Duration) {
```

Comment:
same here, if you pass `now time.Time`, the `Clock` interface is not needed at all throughout this package.
or better - change this method to `SetExpiry(expiry time.Time)` instead of the awkward clock + ttl to reflect the underlying field more correctly.

Reply:
@awly We already have a `SetExpiry` method on all types which simply uses `time.Time`, but `SetTTL` seems to be used by the server with fake clocks and real clocks alike. Not sure if this is a functionality that we want to deprecate in the future, but IIUC, this is out of scope for this PR.

Reply:
It would be just a matter of pulling `clock.Now()` calls up the stack.
The current change allows us to remove `clockwork` from the `/api/go.mod`, but all callers will either have to use `clockwork.Clock` or implement `interface.Clock` manually with `time.Now()`. This feels clunky.

Reply:
@awly @russjones I replaced `clock` with `time.Time` as suggested. The only issue is that we're instantly deprecating the old version of `SetTTL`, `GenerateSchedule`, and `RotationSchedule.CheckAndSetDefaults`.